### PR TITLE
update AliGenPythiaPlus with same way to select calorimeter acceptanc…

### PR DESCRIPTION
**Pull request by gconesab**
…e and particles as in AliGenPythia.
This will allow to use Pythia8 in the same way as Pythia6 to trigger the events on high pT decay photons like in past MC productions.

### CODE REVIEW

**Comment by gconesab** [comment]
Hello, can someone at least do +test?
Here I just copy pasted and adapted what is present already in AliGenPhythia (Pythia6) so that the same tricks can be applied on Pythia8.

**Comment by ktf** [comment]
+test

**Comment by amorsch** [comment]
The changes look reasonable to me.
Andreas

**Comment by gconesab** [comment]
Thanks Andreas, can someone approve +1?

**Comment by sawenzel** [comment]
I have a general remark: It seems to me that many changes are just reformatting of the code which makes it a bit hard to identify the actual things done.

File: PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx [hunk 1]
```diff
@@ -43,10 +32,12 @@
...
 #include "AliRunLoader.h"
 #include "AliMC.h"
+#include "AliLog.h"
 #include "PyquenCommon.h"
 
-ClassImp(AliGenPythiaPlus)
-
+/// \cond CLASSIMP
+ClassImp(AliGenPythiaPlus);
+/// \endcond
```
Comment (sawenzel) [inline PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx]: Is this change necessary? Do we need the `;`?

File: PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx [hunk 2]
```diff
@@ -1381,81 +1339,365 @@ void AliGenPythiaPlus::GetSubEventTime()
...
+///
+/// \param eta        particle pseudorapidity,  etamin=-etamax
+///
+Bool_t AliGenPythiaPlus::IsInBarrel(Float_t eta) const
+{
+  if( eta < fTriggerEta  ) 
+  return kTRUE;
+  else 
+  return kFALSE;
+}
```
Comment (sawenzel) [inline PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx]: consider using `return (eta < fTriggerEta);`

**Comment by pzhristov** [comment]
I have couple of comments:
- in two places I see semicolon after ClassDef or ClassImp. Please remove them, these are Root macros and semicolon would generate warnings on some platforms
- The code below
+Bool_t AliGenPythiaPlus::IsInBarrel(Float_t eta) const
 +{
 +  if( eta < fTriggerEta  ) 
 +  return kTRUE;
 +  else 
 +  return kFALSE;
 +}

can be replaced by a single line 

return (eta < fTriggerEta);

File: PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx (hunk 1 above)
Comment (gconesab) [inline PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx]: This is needed for doxygen, I can remove it.

There are only a couple of reformating, most of the changes are replacing one old piece of code by another.

File: PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx (hunk 2 above)
Comment (gconesab) [inline PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx]: It is a copy paste from AliGenPythia, I can change it here and there or only here.

Comment (sawenzel) [inline PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx]: Thanks. So it seems that we have a lot of common functionality between the 2 classes. The proper thing would probably be to factor out common code or dispatch from one to the other.
[This is something we can keep in mind for AliceO2 ... not asking to completely change the classes now.]

Comment (gconesab) [inline PYTHIA6/AliPythia6/AliGenPythiaPlus.cxx]: I thought of doing this, but preferred simplicity and there are subtleties between the 2 generators that cannot be treated similarly (status code only 0 or 1 in pythia8, label index+1 in pythia8). I will do the requested changes

**Comment by gconesab** [comment]
Requested changes done, please approve or reject.

**Comment by sawenzel** [comment]
+test

**Comment by gconesab** [comment]
hello, can someone approve?

**Comment by pzhristov** [comment]
+1
